### PR TITLE
zscalar enhancement

### PR DESCRIPTION
**Pull request by reutshal**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30741

## Description
- Added the `zscaler-url-quota` command
- Added an argument to the `zscaler-get-categories` command to return only customed categories.
- Added UT

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

### CODE REVIEW

File: Packs/Zscaler/pack_metadata.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "Zscaler",
     "description": "Zscaler is a cloud security solution built for performance and flexible scalability.",
     "support": "xsoar",
-    "currentVersion": "1.0.6",
+    "currentVersion": "1.0.7",
```
Comment (yaakovi) [inline Packs/Zscaler/pack_metadata.json:5]: should be 1.1.0

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.py [hunk 2]
```diff
@@ -823,63 +845,67 @@ def test_module():
...
         return_results(login_command())
-    elif demisto.command() == 'zscaler-logout':
+    elif command == 'zscaler-logout':
         return_results(logout_command())
     else:
         login()
         try:
-            if demisto.command() == 'test-module':
+            if command == 'test-module':
                 demisto.results(test_module())
```
Comment (yaakovi) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.py:860]: lets be consistent, switch all the `demisto.results` to `return_results`

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.py [hunk 3]
```diff
@@ -716,8 +737,9 @@ def get_categories_command(display_url):
     return entry
 
 
-def get_categories():
-    cmd_url = '/urlCategories'
+def get_categories(custom_only=False):
+    cmd_url = '/urlCategories?customOnly=true' if custom_only else '/urlCategories'
```
Comment (yaakovi) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.py:741]: `customOnly=true` should be passed as `params` in the HTTP request and not as part of the URL.
let the requests module do his job

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.py [hunk 4]
```diff
@@ -681,10 +681,31 @@ def category_ioc_update(category_data):
...
-def get_categories_command(display_url):
-    display_urls = True if display_url == 'true' else False
-    raw_categories = get_categories()
+def url_quota_command():
+    cmd_url = '/urlCategories/urlQuota'
+    response = http_request('GET', cmd_url).json()
+
+    human_readable = {
+        'Unique Provisioned URLs': response.get('uniqueUrlsProvisioned'),
+        'Remaining URLs Quota': response.get('remainingUrlsQuota')
```
Comment (yaakovi) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.py:690]:
```suggestion
        'Remaining URLs Quota': response.get('remainingUrlsQuota'),
```

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.py [hunk 5]
```diff
@@ -681,10 +681,31 @@ def category_ioc_update(category_data):
...
+    }
+    entry = {
+        'Type': entryTypes['note'],
+        'Contents': response,
+        'ContentsFormat': formats['json'],
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown("Quota Information", human_readable),
+        'EntryContext': {'Zscaler.Quota': response}
```
Comment (yaakovi) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.py:698]:
```suggestion
        'EntryContext': {'Zscaler.Quota': response},
```

**Review by yaakovi — APPROVED** [review]

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.yml [hunk 6]
```diff
@@ -441,16 +489,33 @@ script:
...
+    execution: false
+    name: zscaler-url-quota
+    outputs:
+    - contextPath: Zscaler.remainingUrlsQuota
+      description: The number of URLs you can add before reaching the quota.
+      type: Number
+    - contextPath: Zscaler.uniqueUrlsProvisioned
+      description: The number of unique URLs that are currently provisioned for your
+        organization.
+      type: Number
```
Comment (yaakovi) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.yml:518]: @amshamah419 please review this part as well

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.yml [hunk 7]
```diff
@@ -346,14 +379,27 @@ script:
...
+      defaultValue: 'false'
+      description: Whether to retrieve only customed categories to the War Room. Default
+        is 'false'.
+      isArray: false
+      name: custom_categories_only
+      predefined:
+      - 'true'
+      - 'false'
+      required: false
+      secret: false
```
Comment (yaakovi) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.yml:402]: @amshamah419 please review this part as well

**Review by yaakovi — APPROVED** [review]

**Comment by reutshal** [comment]
@amshamah419 Can. you please review the yml file (only the parts that @yaakovi mentioned)
And the RN?
Thanks!

File: Packs/Zscaler/Integrations/Zscaler/Zscaler.yml [hunk 8]
```diff
@@ -346,14 +379,27 @@ script:
...
       name: displayURL
       predefined:
       - 'true'
       - 'false'
       required: false
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'false'
+      description: Whether to retrieve only customed categories to the War Room. Default
```
Comment (amshamah419) [inline Packs/Zscaler/Integrations/Zscaler/Zscaler.yml:394]:
```suggestion
      description: Whether to retrieve only custom categories to the War Room. Default
```

File: Packs/Zscaler/ReleaseNotes/1_1_0.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Zscaler
+- Added support for the *custom_categories_only* argument in the ***get-categories*** command.
```
Comment (amshamah419) [inline Packs/Zscaler/ReleaseNotes/1_1_0.md:4]:
```suggestion
- Added support for the *custom_categories_only* argument for the ***get-categories*** command.
```

**Pull request merged**